### PR TITLE
fix: destroy hanging connections after timeout

### PR DESCRIPTION
**Pull request by dignifiedquire**
Ref https://github.com/ipfs/js-ipfs-bitswap/pull/17

### CODE REVIEW

**Comment by diasdavid** [comment]
Can you add a test? (Leave a conn open and see it being destroyed)

File: src/index.js [hunk 1]
```diff
@@ -86,13 +97,22 @@ function TCP () {
   }
 
   this.close = (callback) => {
+    const closeTimeout = 300
```
Comment (diasdavid) [inline src/index.js]: uppercase at the top of the file?

File: src/index.js [hunk 2]
```diff
@@ -56,6 +56,17 @@ function TCP () {
         }
         handler(conn)
       })
+
+      listener.__connections = {}
```
Comment (diasdavid) [inline src/index.js]: why double `_` this time?

File: src/index.js (hunk 1 above)
Comment (dignifiedquire) [inline src/index.js]: we might want to make it configurable per call what do you think?

**Comment by diasdavid** [comment]
Once the test is added (and CI passes) feel free to merge and publish. I have given you publish perm on npm. Need to run now for class, will be available again around 5:30~6pm

File: src/index.js (hunk 2 above)
Comment (dignifiedquire) [inline src/index.js]: cause there is a prop _conncetions from node

**Comment by dignifiedquire** [comment]
Thanks will do

**Pull request merged**